packages/components/steps/src/step.vue

```diff
@@ -60,63 +60,30 @@ import {
 import { ElIcon } from '@element-plus/components/icon'
 import { Close, Check } from '@element-plus/icons'
 
-import type { Ref, PropType, Component } from 'vue'
+import { stepsContextKey } from '@element-plus/tokens/steps'
+import { throwError } from '@element-plus/utils/error'
+import { stepProps } from './step'
+import type { StepsContext } from './step'
 
-export interface IStepsProps {
-  space: number | string
-  active: number
-  direction: string
-  alignCenter: boolean
-  simple: boolean
-  finishStatus: string
-  processStatus: string
-}
-
-export interface StepItemState {
-  uid: number
-  currentStatus: string
-  setIndex: (val: number) => void
-  calcProgress: (status: string) => void
-}
-
-export interface IStepsInject {
-  props: IStepsProps
-  steps: Ref<StepItemState[]>
-}
+const COMPONENT_NAME = 'ElStep'
 
 export default defineComponent({
-  name: 'ElStep',
+  name: COMPONENT_NAME,
   components: {
     ElIcon,
     Close,
     Check,
   },
-  props: {
-    title: {
-      type: String,
-      default: '',
-    },
-    icon: {
-      type: [String, Object] as PropType<string | Component>,
-      default: '',
-    },
-    description: {
-      type: String,
-      default: '',
-    },
-    status: {
-      type: String,
-      default: '',
-      validator: (val: string): boolean =>
-        ['', 'wait', 'process', 'finish', 'error', 'success'].includes(val),
-    },
-  },
+  props: stepProps,
   setup(props) {
+    const parent: StepsContext = inject(stepsContextKey)!
```

Comment:
- We can remove annotation since stepsContextKey has its type.
- Also, please remove `!`